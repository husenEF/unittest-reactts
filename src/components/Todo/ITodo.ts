
export interface Todo {
  name: string,
  status: boolean
  id: number
}
export interface TodoList {
  [index: number]: [Todo]
}
export interface TodoProps {
  todo: Todo
}

