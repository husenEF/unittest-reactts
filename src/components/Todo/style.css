.item-container {
  display: flex;
  flex-direction: "row";
  min-width: 250px;
  margin-top: 10px;
  margin-bottom: 10px;
  border: 1px rgb(249, 251, 253) solid;
  padding: 5px;
}
.item-container span {
  flex-grow: 3;
}

.item-container button {
  cursor: pointer;
  flex-grow: 1;
  background-color: aqua;
  border: none;
}
.item-container button:active {
  background-color: aquamarine;
}
