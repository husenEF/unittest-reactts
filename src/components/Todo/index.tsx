import React, { FC } from "react";

import { TodoProps } from "./ITodo";

const Todo: FC<TodoProps> = (props) => {
  const { name, status, id } = props.todo;
  const text = status ? <del>{name}</del> : name;
  if (name === "") return <p data-testid={`todo-${id}`}>no text</p>;
  // return <p data-testid={`todo-${id}`}>{text}</p>;
  return (
    <div style={{ display: "flex", width: 300, flexDirection: "row" }}>
      <p data-testid={`todo-${id}`} style={{ marginRight: 10 }}>
        {text}
      </p>
      <button>update</button>
    </div>
  );
};

export default Todo;
