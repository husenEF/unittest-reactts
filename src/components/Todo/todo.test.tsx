import React from "react";
import { render, screen, cleanup, fireEvent } from "@testing-library/react";
// import renderer from 'react-test-renderer';
import renderer from "react-test-renderer";

import Todo from "./index";

afterEach(cleanup);

test("render Todo completed", () => {
  const todo = { name: "makan", status: true, id: 1 };

  render(<Todo todo={todo} onClick={(id) => {}} />);
  const elementTodo = screen.getByTestId("todo-1");
  // console.log({ elementTodo });
  expect(elementTodo).toHaveTextContent("makan");
  expect(elementTodo).toContainHTML("del");
  // expect(true).toBe(true);
});

test("render Todo un-completed", () => {
  const todo = { name: "olah raga", status: false, id: 2 };

  render(<Todo todo={todo} onClick={(id) => {}} />);
  const elementTodo = screen.getByTestId("todo-2");
  // console.log({ elementTodo });
  expect(elementTodo).toHaveTextContent("olah raga");
  expect(elementTodo).not.toContainHTML("del");
  // expect(true).toBe(true);
});

test("snapshot Todo", () => {
  const todo = { name: "olah raga", status: true, id: 3 };

  const treeElement = renderer
    .create(<Todo todo={todo} onClick={(id) => {}} />)
    .toJSON();
  expect(treeElement).toMatchSnapshot();
});

test("call onclick btn todo", () => {
  const handleClick = jest.fn();
  const todo = { name: "makan", status: true, id: 1 };

  render(<Todo todo={todo} onClick={handleClick} />);
  fireEvent.click(screen.getByTestId("btn-1"));
  expect(handleClick).toHaveBeenCalledTimes(1);
});
