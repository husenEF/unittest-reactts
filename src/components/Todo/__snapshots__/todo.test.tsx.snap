// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Todo 1`] = `
<p
  data-testid="todo-3"
>
  <del>
    olah raga
  </del>
</p>
`;
